File: src/main/java/com/smarthome/backend/model/Room.java
package com.smarthome.backend.model;

import lombok.Data;
import java.util.List;

/**
 * Repräsentiert einen Raum im Grundriss mit Position, Größe und Eigenschaften.
 */
@Data
public class Room {
    private String id;
    private String name;
    private String icon;
    private Double x;
    private Double y;
    private Double width;
    private Double height;
    private String color; // rgba() Format
    private Double temperature; // Optional, null möglich
    private List<Point> points; // Optional, null möglich
}

File: src/main/node/com/smarthome/backend/model/Room.ts
import type { Position } from "../server/actions/action/Position.js";

export class Room {
  id?: string;
  name?: string;
  icon?: string;
  x?: number;
  y?: number;
  width?: number;
  height?: number;
  color?: string;
  temperature?: number;
  points?: Position[];
  index?: number;

  constructor(init?: Partial<Room>) {
    Object.assign(this, init);
  }
}
